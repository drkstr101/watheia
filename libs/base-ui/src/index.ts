export * from './lib/image/image';
export * from './lib/link/link';
export * from './lib/container/container';
export * from './lib/button/button';
