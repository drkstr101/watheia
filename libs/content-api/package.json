{
  "name": "@watheia/content-api",
  "version": "0.0.1",
  "dependencies": {
    "@iarna/toml": "2.2.5",
    "@swc/helpers": "~0.5.6",
    "@watheia/content-model": "0.0.1",
    "fast-glob": "^3.3.2",
    "js-yaml": "4.1.0",
    "lodash": "4.17.21"
  },
  "type": "commonjs",
  "main": "./src/index.js",
  "typings": "./src/index.d.ts"
}
