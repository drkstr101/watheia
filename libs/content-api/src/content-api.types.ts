import { Model } from '@watheia/content-model';

export interface ContentApi {
  resolve: () => ContentApi;
}

export interface LocalContentSchema {
  rootPath: string;
  models: Model[];
}

export interface MetaTag {
  property: string;
  content: string;
  format: 'property' | 'name';
}

type Context = {
  /** The document/asset file path relative to project directory. */
  filePath: string;
};

export type DocumentContext = Context;
export type AssetContext = Context;
