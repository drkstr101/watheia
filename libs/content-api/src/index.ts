export { LocalContentApi } from './lib/content-api';

export type * from './lib/content-api.types';
