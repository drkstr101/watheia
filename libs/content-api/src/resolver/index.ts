export { withLocalResolver as withLocalContent } from './file-reader';
export { resolveReferences } from './resolve-references';
