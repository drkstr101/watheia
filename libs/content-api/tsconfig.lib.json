{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc",
    "declaration": true,
    "types": ["node"]
  },
  "include": ["package.json", "src/**/*.js", "src/**/*.ts"],
  "exclude": ["jest.config.ts", "spec/**/*.js", "spec/**/*.ts"]
}
