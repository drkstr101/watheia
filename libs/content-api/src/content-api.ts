import { models } from '@watheia/content-model';

import { LocalContentSchema } from './content-api.types';
import { withLocalResolver } from './resolver/file-reader';
import { scanContentDir } from './util/file-utils';

export const defaultOptions = {
  rootPath: process.env['WORKSPACE_ROOT'] ?? process.cwd(),
  models: Object.values(models),
};

export class LocalContentApi {
  public static async create(
    schema: LocalContentSchema = defaultOptions
  ): Promise<LocalContentApi> {
    const filePaths = await scanContentDir(schema.rootPath);
    const documents = await Promise.all(filePaths.map(withLocalResolver(schema)));

    // const fileToContent = Object.fromEntries(documents.map((doc) => [doc.__metadata.id, doc]));
    // return new LocalContentApi(documents.map((doc) => resolveReferences(doc, fileToContent)));
    return new LocalContentApi(documents);
  }

  constructor(public readonly documents: unknown[]) {}
}
