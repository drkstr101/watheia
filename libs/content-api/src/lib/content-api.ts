import type { ContentCache, LocalContentSchema } from './content-api.types';
import { resolveContent, resolveStaticPaths, resolveStaticProps } from './resolver';

export class LocalContentApi {
  public static async create(schema: LocalContentSchema): Promise<LocalContentApi> {
    return new LocalContentApi(await resolveContent(schema));
  }

  constructor(public readonly cache: ContentCache) {}

  resolvePaths(): string[] {
    return resolveStaticPaths(this.cache);
  }

  resolveProps(urlPath: string) {
    return resolveStaticProps(urlPath, this.cache);
  }
}
