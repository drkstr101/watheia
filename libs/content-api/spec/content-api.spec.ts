import { models } from '@watheia/content-model';
import { join } from 'path';

import { LocalContentApi } from '../src/content-api';

const mockOptions = {
  models: Object.values(models),
  rootPath: join(__dirname, '__fixtures__'),
};

describe('watheia.content-api', () => {
  describe('#resolve()', () => {
    it('fully resolves all content entries with the provided schema', async () => {
      const api = await LocalContentApi.create(mockOptions);
      expect(api).toBeInstanceOf(LocalContentApi);
      expect(api.objectsById).toMatchSnapshot();
    });
  });
});
