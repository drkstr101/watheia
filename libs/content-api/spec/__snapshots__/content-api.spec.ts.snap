// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watheia.content-api #resolve() fully resolves all content entries with the provided schema 1`] = `
{
  "content/data/config.json": {
    "favicon": "/images/favicon.svg",
    "footer": {
      "copyrightText": "Powered by Netlify Create.",
      "navLinks": [
        {
          "href": "/",
          "label": "Home",
          "type": "Link",
          "underline": "hover",
        },
        {
          "href": "/about",
          "label": "About",
          "type": "Link",
          "underline": "hover",
        },
        {
          "href": "https://docs.netlify.com/create/overview",
          "label": "Documentation",
          "type": "Link",
          "underline": "hover",
        },
      ],
    },
    "header": {
      "navLinks": [
        {
          "href": "/",
          "label": "Home",
          "type": "Link",
          "underline": "hover",
        },
        {
          "href": "/about",
          "label": "About",
          "type": "Link",
          "underline": "hover",
        },
        {
          "href": "https://docs.netlify.com/create/overview",
          "label": "Documentation",
          "type": "Link",
          "underline": "hover",
        },
      ],
      "title": "TypeScript + MUI Starter",
    },
    "type": "Config",
  },
  "content/data/people/jane-doe.json": {
    "image": {
      "altText": "Jane Doe",
      "src": "/images/jane-doe.jpg",
      "type": "Image",
    },
    "name": "Jane Doe",
    "role": "Co-Founder / CEO",
    "type": "Person",
  },
  "content/data/theme.json": {
    "mode": "light",
    "primaryColor": "#4C57C5",
    "secondaryColor": "#F65458",
    "type": "Theme",
  },
  "content/pages/about.md": {
    "content": "",
    "sections": [
      {
        "subtitle": "This is the subtitle",
        "text": "This is **Markdown** *text*.


Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
",
        "title": "Example for H2 headline",
        "type": "HeroSection",
      },
    ],
    "slug": "/about",
    "title": "About",
    "type": "Page",
  },
  "content/pages/blog/article-1.md": {
    "author": "content/data/people/jane-doe.json",
    "content": "
_View the [source of this content](http://github.github.com/github-flavored-markdown/sample_content.html)._

Let's get the whole "linebreak" thing out of the way. The next paragraph contains two phrases separated by a single newline character:

Roses are red
Violets are blue

The next paragraph has the same phrases, but now they are separated by two spaces and a newline character:

Roses are red
Violets are blue

Oh, and one thing I cannot stand is the mangling of words with multiple underscores in them like perform_complicated_task or do_this_and_do_that_and_another_thing.

## A bit of the GitHub spice

In addition to the changes in the previous section, certain references are auto-linked:

- SHA: be6a8cc1c1ecfe9489fb51e4869af15a13fc2cd2
- User@SHA ref: mojombo@be6a8cc1c1ecfe9489fb51e4869af15a13fc2cd2
- User/Project@SHA: mojombo/god@be6a8cc1c1ecfe9489fb51e4869af15a13fc2cd2
- \\#Num: #1
- User/#Num: mojombo#1
- User/Project#Num: mojombo/god#1

These are dangerous goodies though, and we need to make sure email addresses don't get mangled:

My email addy is <tom@github.com>.

## Math is hard, let's go shopping

In first grade I learned that 5 > 3 and 2 < 7. Maybe some arrows. 1 -> 2 -> 3. 9 <- 8 <- 7.

Triangles man! a^2 + b^2 = c^2

## We all like making lists

The above header should be an H2 tag. Now, for a list of fruits:

- Red Apples
- Purple Grapes
- Green Kiwifruits

Let's get crazy:

1. This is a list item with two paragraphs. Lorem ipsum dolor
   sit amet, consectetuer adipiscing elit. Aliquam hendrerit
   mi posuere lectus.

   Vestibulum enim wisi, viverra nec, fringilla in, laoreet
   vitae, risus. Donec sit amet nisl. Aliquam semper ipsum
   sit amet velit.

2. Suspendisse id sem consectetuer libero luctus adipiscing.

What about some code **in** a list? That's insane, right?

1.  In Ruby you can map like this:

        ['a', 'b'].map { |x| x.uppercase }

2.  In Rails, you can do a shortcut:

        ['a', 'b'].map(&:uppercase)

Some people seem to like definition lists

<dl>
  <dt>Lower cost</dt>
  <dd>The new version of this product costs significantly less than the previous one!</dd>
  <dt>Easier to use</dt>
  <dd>We've changed the product so that it's much easier to use!</dd>
</dl>

## I am a robot

Maybe you want to print \`robot\` to the console 1000 times. Why not?

    def robot_invasion
      puts("robot " * 1000)
    end

You see, that was formatted as code because it's been indented by four spaces.

How about we throw some angle braces and ampersands in there?

    <div class="footer">
        &copy; 2004 Foo Corporation
    </div>

## Set in stone

Preformatted blocks are useful for ASCII art:

<pre>
             ,-.
    ,     ,-.   ,-.
   / \\   (   )-(   )
   \\ |  ,.>-(   )-<
    \\|,' (   )-(   )
     Y ___\`-'   \`-'
     |/__/   \`-'
     |
     |
     |    -hrr-
  ___|_____________
</pre>

## Playing the blame game

If you need to blame someone, the best way to do so is by quoting them:

> I, at any rate, am convinced that He does not throw dice.

Or perhaps someone a little less eloquent:

> I wish you'd have given me this written question ahead of time so I
> could plan for it... I'm sure something will pop into my head here in
> the midst of this press conference, with all the pressure of trying to
> come up with answer, but it hadn't yet...
>
> I don't want to sound like
> I have made no mistakes. I'm confident I have. I just haven't - you
> just put me under the spot here, and maybe I'm not as quick on my feet
> as I should be in coming up with one.

## Table for two

<table>
  <tr>
    <th>ID</th><th>Name</th><th>Rank</th>
  </tr>
  <tr>
    <td>1</td><td>Tom Preston-Werner</td><td>Awesome</td>
  </tr>
  <tr>
    <td>2</td><td>Albert Einstein</td><td>Nearly as awesome</td>
  </tr>
</table>

## Crazy linking action

I get 10 times more traffic from [Google] [1] than from
[Yahoo] [2] or [MSN] [3].

[1]: http://google.com/ 'Google'
[2]: http://search.yahoo.com/ 'Yahoo Search'
[3]: http://search.msn.com/ 'MSN Search'
",
    "date": "2024-01-01",
    "description": "Quibusdam quam omnis voluptatem ipsa omnis magnam dolorem. Nam ut qui nesciunt aliquam et qui numquam.",
    "slug": "article-1",
    "title": "Ullam qui ipsa sapiente et eaque tempore.",
    "type": "Article",
  },
  "content/pages/index.md": {
    "content": "",
    "sections": [
      {
        "actions": [
          {
            "color": "primary",
            "href": "https://docs.netlify.com/create/quickstart",
            "label": "Start Building",
            "size": "lg",
            "type": "Button",
            "variant": "solid",
          },
          {
            "color": "primary",
            "href": "https://docs.netlify.com/create/overview",
            "label": "Read the Docs",
            "size": "lg",
            "type": "Button",
            "variant": "outline",
          },
        ],
        "image": {
          "altText": "Hero section image",
          "src": "/images/hero.svg",
          "type": "Image",
        },
        "subtitle": "Featuring TypeScript, Next.js, MUI v5 & Emotion",
        "text": "This is the HeroSection component. You can visually edit this component & add more sections while developing locally. [Learn more.](https://docs.netlify.com/create/features/local-development)
",
        "title": "Netlify Create Starter",
        "type": "HeroSection",
      },
      {
        "items": [
          {
            "actions": [
              {
                "href": "https://docs.netlify.com/create/overview",
                "label": "Read the Docs",
                "type": "Button",
              },
            ],
            "image": {
              "altText": "First item image",
              "src": "/images/nextjs.svg",
              "type": "Image",
            },
            "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
",
            "title": "First Card Title",
            "type": "Card",
          },
          {
            "actions": [
              {
                "href": "https://docs.netlify.com/create/overview",
                "label": "Read the Docs",
                "type": "Button",
              },
            ],
            "image": {
              "altText": "Second item image",
              "src": "/images/mui-5.svg",
              "type": "Image",
            },
            "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
",
            "title": "Second Card Title",
            "type": "Card",
          },
          {
            "actions": [
              {
                "href": "https://docs.netlify.com/create/overview",
                "label": "Read the Docs",
                "type": "Button",
              },
            ],
            "image": {
              "altText": "Third item image",
              "src": "/images/ts.svg",
              "type": "Image",
            },
            "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
",
            "title": "Third Card Title",
            "type": "Card",
          },
        ],
        "subtitle": "Section subtitle here",
        "title": "Cards Section Component",
        "type": "CardsSection",
      },
    ],
    "slug": "/",
    "title": "Home",
    "type": "Page",
  },
}
`;
